treatments.hpp

```diff
@@ -36,6 +37,27 @@ enum class TreatmentApplication {
     AllInfectedInCell  ///< All infected individuals are removed, rest by ratio
 };
 
+/**
+ * @brief Get treatment application enum from string
+
+ * Throws an std::invalid_argument exception if the value
+ * is not supported.
+ */
+inline TreatmentApplication treatment_app_enum_from_string(const std::string& text)
+{
+    std::map<std::string, TreatmentApplication> mapping{
+        {"ratio_to_all", TreatmentApplication::Ratio},
```

Comment:
I support having alternative versions in the library. I already decided for that in the SI/SEI code: https://github.com/ncsu-landscape-dynamics/PoPS/pull/51/files#diff-f728f36c8a8c4becfad6d804458a3849R56

Each interface can decide what makes sense in its context and the lib can support that by accepting more than one version. This is especially true in case of different spellings (dashes, underscores, case, ...).